rn/Teacher/src/modules/graphql-speed-grader/components/SubmissionPicker.js

```diff
@@ -0,0 +1,164 @@
+
+    if (submission && submission.submissionHistory &&
+      submission.submissionHistory.edges.length > 1) {
+      let selected = submission.submissionHistory.edges[this.props.selectedIndex].submission
+      return <View style={styles.container}>
+        <TouchableHighlight
+          underlayColor='white'
+          onPress={this._togglePicker}
+          testID='header.toggle-submission_history-picker'
+          accessibilityTraits='button'
+        >
+          <View style={styles.submissionHistoryContainer}>
+            <Text style={[styles.submissionDate, this.state.showingPicker && { color: brand.primaryBrandColor }]}>
+              {formattedDueDate(new Date((selected && selected.submittedAt) || ''))}
+            </Text>
+            <Image source={Images.pickerArrow} style={[{ alignSelf: 'center' }, this.state.showingPicker && styles.arrowSelecting]} />
+          </View>
+        </TouchableHighlight>
+        { this.state.showingPicker &&
+          <PickerIOS
+            selectedValue={this.props.selectedIndex}
+            onValueChange={this.changeSelectedSubmission}
+            testID='header.picker'
+          >
+            {submission.submissionHistory.edges.map(({ submission: sub }, idx) => (
+              <PickerItemIOS
+                key={sub.id}
+                value={idx}
+                label={formattedDueDate(new Date(sub.submittedAt || ''))}
+              />
+            ))}
+          </PickerIOS>
+        }
+      </View>
+    } else {
+      if (submission.state === 'unsubmitted') return <View style={[styles.submissionHistoryContainer, styles.noSub]} />
+      return (
+        <View style={styles.container}>
+          <View style={styles.submissionHistoryContainer}>
+            <Text style={styles.submissionDate}>
+              {formattedDueDate(new Date(submission.submittedAt || ''))}
+            </Text>
+          </View>
+        </View>
+      )
+    }
+  }
+}
+
+const styles = StyleSheet.create({
+  container: {
+    marginTop: 16,
+    paddingHorizontal: 16,
+    borderBottomColor: '#D8D8D8',
+    borderBottomWidth: StyleSheet.hairlineWidth,
+    borderStyle: 'solid',
+    paddingBottom: 4,
+  },
+  noSub: {
+    paddingBottom: 21,
+  },
+  submissionHistoryContainer: {
+    flex: 0,
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+  },
+  submissionDate: {
+    color: '#8B969E',
+    fontSize: 14,
+    fontWeight: '500',
+  },
+  arrowSelecting: {
+    transform: [
+      { rotate: '180deg' },
+    ],
+  },
+})
+
+type RouterProps = {
+  submissionID: ?string,
+  submissionProps: SubmissionDataProps,
+}
+
+type State = {
+  showingPicker: boolean,
+}
+
+type SubmissionPickerDataProps = {
+  selectedIndex: ?number,
+}
+
+type SubmissionPickerActionProps = {
+  selectSubmissionFromHistory: Function,
+}
+
+type SubmissionPickerProps
+  = RouterProps
+  & SubmissionPickerDataProps
+  & SubmissionPickerActionProps
```

Comment:
are these types just for reference then, since we don't run `flow`?

Reply:
Yeah I just left them in since we haven't really been removing flow types despite not running it.